### PR TITLE
TINY-6860: Change default height of editor

### PR DESCRIPTION
**Pull request by tiny-ben-tran**
Related Ticket: TINY-6860

Description of Changes:
* Changed the default height of Editor from `200px` to `400px`
* Set the default min height of Editor to 100px so it can't be made disappeared while resizing

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/main/ts/api/Options.ts [hunk 1]
```diff
@@ -274,7 +275,7 @@ const isReadOnly = option('readonly');
 const getHeightOption = option('height');
 const getWidthOption = option('width');
 const getMinWidthOption = wrapOptional(option('min_width'));
-const getMinHeightOption = wrapOptional(option('min_height'));
+const getMinHeightOption = option('min_height');
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/main/ts/api/Options.ts:278]: Why was this changed from being wrapped in an `Optional`? Instead you're now duplication the logic by having to use `Optional.from` everywhere. Please look at reverting this unless there's a good reason it was changed.

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { McEditor } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.themes.silver.editor.DefaultHeight', () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:7]:
```suggestion
describe('browser.tinymce.themes.silver.editor.DefaultHeightTest', () => {
```

File: modules/tinymce/src/themes/silver/main/ts/api/Options.ts (hunk 1 above)
Comment (tiny-ben-tran) [reply modules/tinymce/src/themes/silver/main/ts/api/Options.ts:278]: I don't have a good reason beside trying to be consistent as other options that have a default value. Happy to revert it because it looks less messy

Comment (lnewson) [reply modules/tinymce/src/themes/silver/main/ts/api/Options.ts]: I'm not sure what you mean here by consistent as everything around it does the same wrapping sorry 🤷 Anyways, thanks for putting this back 👍

File: modules/tinymce/CHANGELOG.md [hunk 3]
```diff
@@ -31,6 +31,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Notifications no longer require a timeout to disable the close button #TINY-6679
 - The editor theme is now fetched in parallel with the icons, language pack and plugins #TINY-8453
 - Calls to `editor.selection.setRng` now update the cursor position bookmark used when focus is returned to the editor #TINY-8450
+- Changed the default height of Editor from `200px` to `400px` #TINY-6860
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:34]: This should probably be in the `Improved` section, as it's being changed to make the editor more usable by default. Also `Editior` shouldn't be capitalised.

- [ ] Move this to Improved and reword to match

File: modules/tinymce/CHANGELOG.md [hunk 4]
```diff
@@ -31,6 +31,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Set the default min height of Editor to 100px so it can't be made disappeared while resizing  #TINY-6860
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:35]:
```suggestion
-  The default minimum height of editor has been changed to 100px to prevent the UI disappearing while resizing  #TINY-6860
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    base_url: '/project/tinymce/js/tinymce'
+  };
+
+  it('TINY-8443: Default height should be 400px', async () => {
+    const editor = await McEditor.pFromSettings<Editor>(baseSettings);
+    assert.equal(editor.getContainer().offsetHeight, 400);
+    McEditor.remove(editor);
+  });
+
+  it('TINY-8443: Option heigh takes precedence', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:18]:
```suggestion
  it('TINY-8443: Height option takes precedence over the default', async () => {
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    const editor = await McEditor.pFromSettings<Editor>({ height: 500, ...baseSettings });
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:19]:
```suggestion
    const editor = await McEditor.pFromSettings<Editor>({ ...baseSettings, height: 500 });
```
If they are base settings, then they shouldn't attempt to be overriding anything specified, so the base settings should be spread first.

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    assert.equal(editor.getContainer().offsetHeight, 500);
+    McEditor.remove(editor);
+  });
+
+  it('TINY-8443: CSS height takes precendence in the absence of option height', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:24]:
```suggestion
  it('TINY-8443: CSS height takes precedence in the absence of the height option when larger than the default height', async () => {
```
This needs more clarification, as it shouldn't take precedence if the CSS height is less than 400px.

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    let editor = await McEditor.pFromHtml<Editor>('<div style="height: 600px"></div>', baseSettings);
+    assert.equal(editor.getContainer().offsetHeight, 600);
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:26]: This assertion is repeated a lot here... can we pull it out into a helper function? e.g.
```
const assertEditorHeight = (editor: Editor, expected: number) =>
  assert.equal(editor.getContainer().offsetHeight, expected);
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    McEditor.remove(editor);
+
+    // Test with another height unit
+    editor = await McEditor.pFromHtml<Editor>('<div style="height: 50rem"></div>', baseSettings);
+    assert.equal(editor.getContainer().offsetHeight, 800);
+    McEditor.remove(editor);
+  });
+
+  it('TINY-8443: Use default height if CSS height is less and option height is not specified', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:35]:
```suggestion
  it('TINY-8443: Use default height if CSS height is less and the height option is not specified', async () => {
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    const editor = await McEditor.pFromHtml<Editor>('<div style="height: 200px"></div>', baseSettings);
+    assert.equal(editor.getContainer().offsetHeight, 400);
+    McEditor.remove(editor);
+  });
+
+  it('TINY-8443: Option height should take precedence over CSS height', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:41]:
```suggestion
  it('TINY-8443: The height option should take precedence over CSS height', async () => {
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    const editor = await McEditor.pFromHtml<Editor>('<div style="height: 600px"></div>', { height: 721, ...baseSettings });
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:42]:
```suggestion
    const editor = await McEditor.pFromHtml<Editor>('<div style="height: 600px"></div>', { ...baseSettings, height: 721 });
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    assert.equal(editor.getContainer().offsetHeight, 721);
+    McEditor.remove(editor);
+  });
+
+  it('TINY-8443: Default min-height to 100px', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:47]:
```suggestion
  it('TINY-8443: The min-height option defaults to 100px', async () => {
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    const editor = await McEditor.pFromSettings<Editor>({ height: 80, ...baseSettings });
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/DefaultHeightTest.ts:48]:
```suggestion
    const editor = await McEditor.pFromSettings<Editor>({ ...baseSettings, height: 80 });
```

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/NotificationManagerImplTest.ts [hunk 14]
```diff
@@ -114,21 +114,21 @@ describe('browser.tinymce.themes.silver.editor.NotificationManagerImplTest', ()
       assertStructure('Check success notification structure', nSuccess, 'success', 'Message 4');
 
       // Check items are positioned so that they are stacked
-      assertPosition('Error notification', nError, 220, -200);
-      assertPosition('Warning notification', nWarn, 220, -152);
-      assertPosition('Info notification', nInfo, 220, -104);
-      assertPosition('Success notification', nSuccess, 220, -56);
+      assertPosition('Error notification', nError, 220, -400);
```
Comment (lnewson) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/NotificationManagerImplTest.ts:117]: I wonder if instead of these tests relying on the default height if we should specify it in the init configuration so it won't change (like we do with the width)?

**Review by lnewson — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 3 above)
Comment (tiny-ben-tran) [reply modules/tinymce/CHANGELOG.md:34]: It is already in Improved section

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:34]: Ahh sorry, it started with `Changed` and I couldn't see the top. In that case the line below should be in `Changed` instead and this needs rewording.

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/NotificationManagerImplTest.ts (hunk 14 above)
Comment (tiny-ben-tran) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/NotificationManagerImplTest.ts]: I set the height to 200 and reverted other changes. It looks cleaner.

File: modules/tinymce/CHANGELOG.md [hunk 15]
```diff
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Notifications no longer require a timeout to disable the close button #TINY-6679
 - The editor theme is now fetched in parallel with the icons, language pack and plugins #TINY-8453
 - Calls to `editor.selection.setRng` now update the cursor position bookmark used when focus is returned to the editor #TINY-8450
+- The default minimum height of editor has been changed to 100px to prevent the UI disappearing while resizing #TINY-6860
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:34]: This is the opposite of what I said needed doing... this should be changed and the other should be in Improved.

Comment (tiny-ben-tran) [reply modules/tinymce/CHANGELOG.md:34]: Sorry. I didn't read it carefully. 
However, makes more sense to me for this to be in the Improved section since it improves the usability of the UI.

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:34]: It's not an improvement, if anything it's a bug fix, but because it's changing behaviour the `Changed` is a higher precedence so it should go in there.

File: modules/tinymce/CHANGELOG.md [hunk 16]
```diff
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Notifications no longer require a timeout to disable the close button #TINY-6679
 - The editor theme is now fetched in parallel with the icons, language pack and plugins #TINY-8453
 - Calls to `editor.selection.setRng` now update the cursor position bookmark used when focus is returned to the editor #TINY-8450
+- The default height of editor is changed from `200px` to `400px` #TINY-6860
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:34]: As I said you can't use `changed` in an improved entry, so how about something like this:
```suggestion
- The default height of editor has been increased from `200px` to `400px` #TINY-6860
```

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:34]: I am wondering if we need to say why though, as remember this is a UX improvement to make the editor more usable since 200px was deemed too small for content authoring.

Comment (tiny-ben-tran) [reply modules/tinymce/CHANGELOG.md:34]: Going to change it `The default height of editor has been increased from `200px` to `400px` to improve the usability of UI #TINY-6860`?

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:34]: Sounds good 👍

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md:34]: actually, maybe say `usability of the editor` rather than the UI?

**Review by ltrouton — APPROVED** [review]

**Review by TheSpyder — APPROVED** [review]
